cmds/xinit/xinit.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+var (
+	tczPackages  = []string{"Xorg-7.7-bin", "Xorg-7.7-dev", "Xorg-7.7-lib-dev", "Xorg-7.7-lib", "Xorg-7.7", "Xorg-fonts", "xorg-proto", "xorg-server-dev",
+"xorg-server","Xprogs","Xlibs","xpad-locale","xpad","Xlibs","Xfbdev","xvid","xkeyboard-config","xz","gdb","libX11-dev","libX11","libXfixes","i2c-4.8.17-tinycore64","graphics-4.8.17-tinycore64","libXvmc","libdrm","pixman","libXinerama","libXrandr","libXdamage","libXcursor","libXtst","libxshmfence","xf86-video-intel","openssh","aterm"}
+	sshdCommands = []string{"Protocol 2", "AcceptEnv LANG LC_*", "UsePAM no", "ChallengeResponseAuthentication no", "passwordauthentication no", "AuthorizedKeysFile ~/.ssh/authorized_keys", "PermitRootLogin without-password", "X11Forwarding yes", "RSAAuthentication yes", "PubkeyAuthentication yes", "X11DisplayOffset 10", "X11UseLocalhost yes"}
+)
+
+func setup() error {
+	if false {
+		return syscall.Mount("/tmp", "/tmp", "tmpfs", syscall.MS_MGC_VAL, "")
+	}
+	cmd := exec.Command("dhclient", "&")
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = os.Stdin, os.Stdout, os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		log.Printf(" error: %v. Continuing...", err)
+	}
+	fmt.Printf("Ip link below: \n")
+	cmd = exec.Command("ip", "link", "show")
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = os.Stdin, os.Stdout, os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		log.Printf(" error: %v. Continuing...", err)
+	}
+	return nil
+}
+
+func sshSetup() error {
+	sshdLoc := "/etc/ssh/sshd_config"
+	cmd := exec.Command("tcz", "openssh")
+	cmd.Stdin, cmd.Stdout, cmd.Stderr = os.Stdin, os.Stdout, os.Stderr
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	//cp /etc/ssh/sshd_config /etc/ssh/sshd_config_backup
+	if _, err := os.Stat(sshdLoc); err != nil {
+		return err
+	}
+	fileContent, err := ioutil.ReadFile(sshdLoc)
```

Comment:
If this file happens to be really big, this would read a lot of stuff into memory. A nicer way to do this would be to use `io.Copy`, which maintains a smaller buffer.

```go
sshdFile, err := os.Open(sshdLoc)
...
sshFileTarget, err := os.OpenFile(sshdLoc + "_backup", os.RDWR|os.CREATE, 0777)
...
_, err := io.Copy(sshFileTarget, sshFile)
...
```